components-compose/src/main/java/uk/gov/hmrc/components/compose/extensions/HtmlTextExt.kt

```diff
@@ -0,0 +1,244 @@
+    }.toAnnotatedString(urlSpanStyle, colorMapping)
+
+    HtmlText(
+        modifier = modifier,
+        annotatedString = annotatedString,
+        color = color,
+        fontSize = fontSize,
+        fontStyle = fontStyle,
+        fontWeight = fontWeight,
+        fontFamily = fontFamily,
+        letterSpacing = letterSpacing,
+        textDecoration = textDecoration,
+        textAlign = textAlign,
+        lineHeight = lineHeight,
+        overflow = overflow,
+        softWrap = softWrap,
+        maxLines = maxLines,
+        inlineContent = inlineContent,
+        onTextLayout = onTextLayout,
+        style = style
+    )
+}
+
+@Composable
+fun HtmlText(
+    modifier: Modifier = Modifier,
+    annotatedString: AnnotatedString,
+    color: Color = Color.Unspecified,
+    fontSize: TextUnit = TextUnit.Unspecified,
+    fontStyle: FontStyle? = null,
+    fontWeight: FontWeight? = null,
+    fontFamily: FontFamily? = null,
+    letterSpacing: TextUnit = TextUnit.Unspecified,
+    textDecoration: TextDecoration? = null,
+    textAlign: TextAlign? = null,
+    lineHeight: TextUnit = TextUnit.Unspecified,
+    overflow: TextOverflow = TextOverflow.Clip,
+    softWrap: Boolean = true,
+    maxLines: Int = Int.MAX_VALUE,
+    inlineContent: Map<String, InlineTextContent> = mapOf(),
+    onTextLayout: (TextLayoutResult) -> Unit = {},
+    style: TextStyle = LocalTextStyle.current
+) {
+    val clickable =
+        annotatedString.getStringAnnotations(0, annotatedString.length - 1).any { it.tag == "url" }
+
+    val uriHandler = LocalUriHandler.current
+    val layoutResult = remember { mutableStateOf<TextLayoutResult?>(null) }
+
+    val urls = remember(layoutResult, annotatedString) {
+        annotatedString.getStringAnnotations("url", 0, annotatedString.lastIndex)
+    }
+
+    Text(
+        modifier = modifier.then(
+            if (clickable) Modifier
+                .pointerInput(Unit) {
+                    detectTapGestures(onTap = { pos ->
+                        layoutResult.value?.let { layoutResult ->
+                            val position = layoutResult.getOffsetForPosition(pos)
+                            annotatedString.getStringAnnotations(position, position).firstOrNull() ?.let { sa ->
+                                if (sa.tag == "url") { uriHandler.openUri(sa.item) }
+                            }
+                        }
+                    })
+                }
+                .semantics {
+                    if (urls.size == 1) {
+                        role = Role.Button
+                        onClick("Link (${annotatedString.substring(urls[0].start, urls[0].end)}") {
+                            uriHandler.openUri(urls[0].item)
+                            true
+                        }
+                    } else {
+                        customActions = urls.map {
+                            CustomAccessibilityAction("Link (${annotatedString.substring(it.start, it.end)})") {
```

Comment:
We don't want any hardcoded text in the component library! (incase we need to support translations, etc.)